apstra/enum.go

```diff
@@ -211,6 +211,42 @@ var (
 		ResourcePoolTypeVlan,
 		ResourcePoolTypeVni,
 	)
+	_                       enum = new(CollectorOSType)
+	CollectorOSTypeJunos         = CollectorOSType{Value: "junos"}
+	CollectorOSTypeJunosEvo      = CollectorOSType{Value: "junos_evo"}
+	CollectorOSTypes             = oenum.New(
+		CollectorOSTypeJunos,
+		CollectorOSTypeJunosEvo,
+	)
+	_                              enum = new(CollectorOSVariant)
+	CollectorOSVariantACX               = CollectorOSVariant{Value: "acx"}
+	CollectorOSVariantACX_F             = CollectorOSVariant{Value: "acx-f"}
+	CollectorOSVariantACX_QFX_7K        = CollectorOSVariant{Value: "acx-qfx-7k"}
+	CollectorOSVariantPTX               = CollectorOSVariant{Value: "ptx"}
+	CollectorOSVariantPTX1K             = CollectorOSVariant{Value: "ptx1k"}
+	CollectorOSVariantQFX_MS_FIXED      = CollectorOSVariant{Value: "qfx-ms-fixed"}
+	CollectorOSVariantJunos             = CollectorOSVariant{Value: "junos"}
+	CollectorOSVariantJunos_EX          = CollectorOSVariant{Value: "junos-ex"}
+	CollectorOSVariantJunos_QFX         = CollectorOSVariant{Value: "junos-qfx"}
+
+	CollectorOSVariants = oenum.New(
+		CollectorOSVariantACX,
+		CollectorOSVariantACX_F,
+		CollectorOSVariantACX_QFX_7K,
+		CollectorOSVariantPTX,
+		CollectorOSVariantPTX1K,
+		CollectorOSVariantQFX_MS_FIXED,
+		CollectorOSVariantJunos,
+		CollectorOSVariantJunos_EX,
+		CollectorOSVariantJunos_QFX,
+	)
+
+	CollectorOSVersion22_2r2 = CollectorOSVersion{Value: "22.2r2"}
+	CollectorOSVersion23_2r1 = CollectorOSVersion{Value: "23.2r1"}
```

Comment:
Version strings probably shouldn't be enums.

Reply:
The collectors seem to support only a certain set of OS versions in the UI (it's a drop-down box).
That's why I made these enums.

Reply:
Interesting.

I just poked around a bit, found that:
- the REST API explorer doesn't mention that this is an enum
- neither of these strings appear anywhere in the controller node under `/usr/lib/python2.7/dist-packages/aos`
- the API will accept other version strings

This is apparently a frontend-only validation? I wonder what's going on here.

My main concern about it are:
- we'll wind up chasing version strings with new Apstra/Junos releases
- if these versions *do* really matter, updating the list won't cut it - we'll need to validate the enums against the in-use Apstra version to really get this right

The secondary concern is that there's not a great way to embed "22.2r2" in a variable name without running afoul of the "no underscores" naming convention.

Reply:
+1 I was going to try random strings to mess with the API, glad you tried it :-)
I changed this to a string.